test/integration/services/dealer/hedge-price.spec.ts

```diff
@@ -0,0 +1,1892 @@
+import { baseLogger } from "@services/logger"
+import { AccountsRepository } from "@services/mongoose"
+
+import {
+  cancelOkexPricePublish,
+  createAndFundNewWalletForPhone,
+  getBalanceHelper,
+  publishOkexPrice,
+  randomPhone,
+} from "test/helpers"
+
+jest.mock("@config", () => {
+  const config = jest.requireActual("@config")
+  return {
+    ...config,
+    getInvoiceCreateAttemptLimits: jest.fn().mockReturnValue({
+      ...config.getInvoiceCreateAttemptLimits(),
+      points: 1000,
+    }),
+  }
+})
+
+const calc = AmountCalculator()
+const usdHedgeEnabled = getDealerConfig().usd.hedgingEnabled
+
+const USD_STARTING_BALANCE = 100 as UsdCents
+
+const ONE_CENT = { amount: 1n, currency: WalletCurrency.Usd } as UsdPaymentAmount
+const ONE_SAT = { amount: 1n, currency: WalletCurrency.Btc } as BtcPaymentAmount
+
+type AccountAndWallets = {
+  newBtcWallet: Wallet
+  newUsdWallet: Wallet
+  newAccount: Account
+}
+
+const btcAmountFromUsdNumber = async (
+  centsAmount: number | bigint,
+): Promise<BtcPaymentAmount> => {
+  const usdPaymentAmount = paymentAmountFromNumber({
+    amount: Number(centsAmount),
+    currency: WalletCurrency.Usd,
+  })
+  if (usdPaymentAmount instanceof Error) throw usdPaymentAmount
+
+  const midPriceRatio = await getMidPriceRatio(usdHedgeEnabled)
+  if (midPriceRatio instanceof Error) throw midPriceRatio
+  return midPriceRatio.convertFromUsd(usdPaymentAmount)
+}
+
+const usdFundingAmount = paymentAmountFromNumber({
+  amount: USD_STARTING_BALANCE,
+  currency: WalletCurrency.Usd,
+})
+if (usdFundingAmount instanceof Error) throw usdFundingAmount
+
+const newAccountAndWallets = async () => {
+  const phone = randomPhone()
+  const newBtcWallet = await createAndFundNewWalletForPhone({
+    phone,
+    balanceAmount: await btcAmountFromUsdNumber(usdFundingAmount.amount),
+  })
+
+  const newUsdWallet = await createAndFundNewWalletForPhone({
+    phone,
+    balanceAmount: usdFundingAmount,
+  })
+
+  const newAccount = await AccountsRepository().findById(newBtcWallet.accountId)
+  if (newAccount instanceof Error) throw newAccount
+
+  return { newBtcWallet, newUsdWallet, newAccount }
+}
+
+const getBtcForUsdEquivalentIntraledger = async ({
+  btcPaymentAmount,
+  accountAndWallets,
+}: {
+  btcPaymentAmount: BtcPaymentAmount
+  accountAndWallets: AccountAndWallets
+}): Promise<CurrencyBaseAmount | ZeroAmountForUsdRecipientError> => {
+  const { newBtcWallet, newUsdWallet, newAccount } = accountAndWallets
+  const sendArgs = {
+    recipientWalletId: newUsdWallet.id,
+    memo: null,
+    senderWalletId: newBtcWallet.id,
+    senderAccount: newAccount,
+  }
+
+  const beforeBtc = await getBalanceHelper(newBtcWallet.id)
+
+  const result = await Payments.intraledgerPaymentSendWalletId({
+    amount: Number(btcPaymentAmount.amount),
+    ...sendArgs,
+  })
+  if (result instanceof Error) {
+    if (!(result instanceof ZeroAmountForUsdRecipientError)) throw result
```

Comment:
why is only this error being thrown?

Reply:
Only the `ZeroAmountForUsdRecipientError` is not being thrown because we'd like to use that value outside this function. All other errors are thrown here like we usually do in tests to fail the test.

Reply:
Fixed: https://github.com/GaloyMoney/galoy/pull/1846/commits/4ebd8ddd7b4fc84653a2c1d96c36044d73429e30